### ISSUE TITLE
textual differences in the disassembly can confuse diaphora

### ISSUE THREAD

**Issue by shuffle2** [joxeankoret/diaphora#19]
for example, within a larger function, there is:
```
stwu      r1, -0x28(r1) # Alternative name is '..bof.obj.5Cghs.5Ccafe.5Ccos.5Cloader.5Ctarget.5CNDEBUG.5Creloc...73rc.5Ccos.5Cloader.5Ctarget..5061DCFB..0'
stmw      r25, 0x28+var_1C(r1)
mr        r28, r9
mflr      r0
lwz       r9, 0x28+arg_C(r1)
mr        r27, r8
lwz       r12, 0x28+arg_10(r1)
lwz       r11, 0x28+arg_8(r1)
stw       r0, 0x28+arg_4(r1)
li        r25, 0
add       r8, r4, r7
lis       r26, -7 # 0xFFF8D3E8
cmpwi     r3, 0
mr        r29, r8
mr        r30, r25
addi      r26, r26, -0x2C18 # 0xFFF8D3E8
beq       loc_1007CA4
```
being compared to:
```
stwu      r1, -0x28(r1) # Alternative name is '..bof.obj.5Cghs.5Ccafe.5Ccos.5Cloader.5Ctarget.5CNDEBUG.5Creloc...73rc.5Ccos.5Cloader.5Ctarget..505FF058..0'
stmw      r25, 0x28+var_1C(r1)
mr        r28, r9
mflr      r0
lwz       r9, 0x28+arg_C(r1)
mr        r27, r8
lwz       r12, 0x28+arg_10(r1)
lwz       r11, 0x28+arg_8(r1)
stw       r0, 0x28+arg_4(r1)
li        r25, 0
add       r8, r4, r7
lis       r26, -7 # 0xFFF8D3E8
cmpwi     r3, 0
mr        r29, r8
mr        r30, r25
addi      r26, r26, -0x2C18 # 0xFFF8D3E8
beq       loc_1007CA4
```
in other words, the only difference is `5061DCFB` vs. `505FF058`, which is some token that the linker has emitted just for it's own house keeping. this causes diaphora to rank quite a few functions as "partial matches" when they are in fact exactly the same.

p.s. there is a similar issue with:
```
bl        __sFree_static_in_DJenkinsworkspace2_07_BUILDsdk_2_07systemobjghscafecosruntimetinyheapNDEBUGTinyHeap_inf
```
vs.
```
bl        __sFree_static_in_DJenkinsworkspaceUPD207_BUILDupdater_2_07systemobjghscafecosruntimetinyheapNDEBUGTinyHeap_inf
```
here a substring of the function name being called differs (was built from a different directory). But I have used "ignore all function names", and diaphora still flags it as a difference.

**Comment by joxeankoret** [comment]
Hi,

Can you share the IDBs or binaries? I think this is easy to partially fix, but I cannot do it without an example. You can use the address and PGP key here: http://joxeankoret.com/contact.html

**Comment by joxeankoret** [comment]
Fixed with commit 81f6fc257b0b4c2bbd74340cff570b7ac3b42bdc
